src/java-interop/java-interop-gc-bridge-mono.c

```diff
@@ -155,11 +160,36 @@ ji_realpath (const char *path)
 	if (path == NULL)
 		return NULL;
 
+#ifndef WINDOWS
 	char   *rp = realpath (path, NULL);
 	if (rp == NULL) {
 		return strdup (path);
 	}
 	return rp;
+#else
+#define BUFSIZE 4096
+
+	wchar_t *wpath = utf8_to_utf16 (path);
+	char *fullpath;
+	wchar_t buffer [BUFSIZE];
```

Comment:
I don't think that a stack-allocated 4K buffer is a good idea... :-/

Reply:
It is pretty common pattern to avoid unnecessary heap allocations, it is even used in the sample in the documentation of the `GetFullNamePath` function. https://msdn.microsoft.com/en-us/library/windows/desktop/aa364963(v=vs.85).aspx